Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/distance.h

```diff
@@ -1310,8 +1313,433 @@ double approximate_symmetric_Hausdorff_distance(const TriangleMesh& tm1,
     tm1, tm2, parameters::all_default(), parameters::all_default());
 }
 
+////////////////////////////////////////////////////////////////////////
+
+namespace internal {
+
+template <class Concurrency_tag,
+          class Kernel,
+          class TriangleMesh,
+          class VPM1,
+          class VPM2>
+double bounded_error_Hausdorff_impl(
+  const TriangleMesh& tm1,
+  const TriangleMesh& tm2,
+  const typename Kernel::FT& error_bound,
+  VPM1 vpm1,
+  VPM2 vpm2)
+{
+  CGAL_assertion_code(  bool is_triangle = is_triangle_mesh(tm1) && is_triangle_mesh(tm2) );
+  CGAL_assertion_msg (is_triangle,
+        "One of the meshes is not triangulated. Distance computing impossible.");
+
+  typedef AABB_face_graph_triangle_primitive<TriangleMesh, VPM1> TM1_primitive;
+  typedef AABB_face_graph_triangle_primitive<TriangleMesh, VPM2> TM2_primitive;
+  typedef AABB_tree< AABB_traits<Kernel, TM1_primitive> > TM1_tree;
+  typedef AABB_tree< AABB_traits<Kernel, TM2_primitive> > TM2_tree;
+  typedef typename AABB_tree< AABB_traits<Kernel, TM2_primitive> >::AABB_traits Tree_traits;
+  typedef typename Tree_traits::Point_and_primitive_id Point_and_primitive_id;
+
+  typedef typename Kernel::Point_3 Point_3;
+  typedef typename Kernel::Triangle_3 Triangle_3;
+
+  typedef typename boost::graph_traits<TriangleMesh>::vertex_descriptor vertex_descriptor;
+  typedef typename boost::graph_traits<TriangleMesh>::face_descriptor face_descriptor;
+  typedef typename boost::graph_traits<TriangleMesh>::halfedge_descriptor halfedge_descriptor;
+
+  typedef std::pair<double, double> Hausdorff_bounds;
+  typedef CGAL::Spatial_sort_traits_adapter_3<Kernel,VPM1> Search_traits_3;
+  typedef CGAL::dynamic_vertex_property_t<std::pair<double, face_descriptor>> Vertex_property_tag;
+  typedef CGAL::dynamic_face_property_t<Hausdorff_bounds> Face_property_tag;
+
+  typedef typename boost::property_map<TriangleMesh, Vertex_property_tag>::const_type Vertex_closest_triangle_map;
+  typedef typename boost::property_map<TriangleMesh, Face_property_tag>::const_type Triangle_hausdorff_bounds;
+
+  typedef std::pair<double, double> Hausdorff_bounds;
+
+  typename Kernel::Compute_squared_distance_3 squared_distance;
+  typename Kernel::Construct_projected_point_3 project_point;
+  typename Kernel::FT dist;
+
+  typedef
+  #if BOOST_VERSION >= 105000
+        boost::heap::priority_queue< Candidate_triangle<Kernel>, boost::heap::compare< std::greater<Candidate_triangle<Kernel> > > >
+  #else
+        std::priority_queue< Candidate_triangle<Kernel> >
+  #endif
+        Heap_type;
+
```

Comment:
I think we should think about adding a preprocessing step (activated with a named parameter?) that checks discard all triangles that are identical is both meshes to avoid doing unnecessary tests.

Reply:
Done.